newIDE/app/src/Utils/GDevelopServices/Authentication.js

```diff
@@ -169,6 +171,44 @@ export default class Authentication {
     });
   };
 
+  changeEmail = (
+    getAuthorizationHeader: () => Promise<string>,
+    form: ChangeEmailForm
+  ) => {
+    return updateEmail(this.firebaseUser, form.email)
+      .then(() => {
+        console.log('Email successfully changed in Firebase');
+        return getAuthorizationHeader();
+      })
+      .then(authorizationHeader => {
+        if (!this.firebaseUser) {
+          console.error('Cannot edit user if not logged in');
+          throw new Error('Cannot edit user if not logged in');
```

Comment:
Nit: period at the end of the sentence and also because this is an error message that is not translated, I would be very precise: "Cannot finish editing the user email address as the user is not logged in anymore."

Because when the time will come for this error to appear, we will be very glad to have something that is very precise about what happened.